src/librustc_typeck/astconv.rs

```diff
@@ -1225,10 +1225,19 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
 
     /// Transform a `PolyTraitRef` into a `PolyExistentialTraitRef` by
     /// removing the dummy `Self` type (`trait_object_dummy_self`).
-    fn trait_ref_to_existential(&self, trait_ref: ty::TraitRef<'tcx>)
-                                -> ty::ExistentialTraitRef<'tcx> {
+    fn trait_ref_to_existential(
+        &self,
+        trait_ref: ty::TraitRef<'tcx>,
+    ) -> ty::ExistentialTraitRef<'tcx> {
```

Comment:
As per my other comment, I think this should just be a closure, and use `dummy_self` (which is in scope where this method is used) instead of `tcx.types.trait_object_dummy_self`.